test/e2e/e2e.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"github.com/onsi/ginkgo"
+	"github.com/rancher/submariner/test/e2e/framework"
+	_ "github.com/rancher/submariner/test/e2e/example"
+)
+
+
+// There are certain operations we only want to run once per overall test invocation
+// (such as deleting old namespaces, or verifying that all system pods are running.
+// Because of the way Ginkgo runs tests in parallel, we must use SynchronizedBeforeSuite
+// to ensure that these operations only run on the first parallel Ginkgo node.
+//
+// This function takes two parameters: one function which runs on only the first Ginkgo node,
+// returning an opaque byte array, and then a second function which runs on all Ginkgo nodes,
+// accepting the byte array.
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
+	// Run only on Ginkgo node 1
+
+	// Wait for readiness of registered clusters to ensure tests
+	// run against a healthy federation.
+	//framework.WaitForUnmanagedClusterReadiness()
+
+	return nil
+
+}, func(data []byte) {
+	// Run on all Ginkgo nodes
+})
+
+// Similar to SynchornizedBeforeSuite, we want to run some operations only once (such as collecting cluster logs).
+// Here, the order of functions is reversed; first, the function which runs everywhere,
+// and then the function that only runs on the first Ginkgo node.
+
+var _ = ginkgo.SynchronizedAfterSuite(func() {
+	// Run on all Ginkgo nodes
+
+	//framework.Logf("Running AfterSuite actions on all node")
+	framework.RunCleanupActions()
+}, func() {
+	// Run only Ginkgo on node 1
+})
```

Comment:
How about merging all of this into e2e_test.go?

Reply:
ok but there's also ginkgo-specific setup in e2e_test.go. I believe other repos define a RunE2ETests function in e2e.go and call it from TestE2E in e2e_test.go. It seems we should follow that pattern as well to maintain the separation of concerns.

Reply:
I will extract that as a RunE2ETests into e2e.go as kubernetes has, not a big deal though.